flow/layers/color_filter_layer_unittests.cc

```diff
@@ -94,17 +101,24 @@ TEST_F(ColorFilterLayerTest, SimpleFilter) {
   EXPECT_TRUE(layer->needs_painting(paint_context()));
   EXPECT_EQ(mock_layer->parent_matrix(), initial_transform);
 
-  SkPaint filter_paint;
-  filter_paint.setColorFilter(layer_filter);
-  layer->Paint(paint_context());
-  EXPECT_EQ(
-      mock_canvas().draw_calls(),
-      std::vector({MockCanvas::DrawCall{
-                       0, MockCanvas::SaveLayerData{child_bounds, filter_paint,
-                                                    nullptr, 1}},
-                   MockCanvas::DrawCall{
-                       1, MockCanvas::DrawPathData{child_path, child_paint}},
-                   MockCanvas::DrawCall{1, MockCanvas::RestoreData{0}}}));
+  DisplayListBuilder expected_builder;
+  /* ColorFilterLayer::Paint() */ {
+    DlPaint dl_paint;
+    dl_paint.setColorFilter(dl_color_filter.get());
+    expected_builder.saveLayer(&child_bounds, &dl_paint);
+    {
+      /* MockLayer::Paint() */ {
+        DlPaint dl_paint;
```

Comment:
Does the compiler allow this without warning about hiding a variable in another scope? It would be better to name them specific to their use. save_paint for the outer one draw_paint for the inner.

Can the drawPath be written as `drawPath(child_path, DlPaint().setColor(...));`?

Reply:
(Inlining wouldn't work for the saveLayer because that method takes a pointer rather than a reference)

Reply:
done